website/blog/spark-dataset-statistics/index.mdx

```diff
@@ -0,0 +1,201 @@
+---
+title: Capturing dataset statistics in Apache Spark
+date: 2024-12-20
+authors: [Leszczynski]
+description: Capturing dataset statistics in Apache Spark
+---
+
+OpenLineage events enable the creation of a consistent lineage graph, where dataset vertices are connected through the jobs that read from and write to them. This graph becomes even more valuable when its nodes and edges are enriched with additional metadata for practical use cases. One important aspect to capture is the amount of data processed, as it facilitates applications such as cost estimation and data quality monitoring, among others.
+In this post, we introduce recent developments in Spark dataset statistics collection and reporting within OpenLineage events. We outline the basic statistics included, as well as the detailed scan and commit reports generated when processing Iceberg datasets.
+
+<!--truncate-->
+
+## Input and output facets
+
+OpenLineage events are made of three main building blocks: runs, jobs, and datasets. Unfortunately,
+metadata about data processed is both run and dataset property.
+On one hand, it is a property of a run. On the other hand, it makes
+sense only in the context of a given dataset. For this purpose, `inputFacets` and `outputFacets` were attached
+to datasets, although logically they describe dataset in the context of a given run.
+
+## Basic input and output statistics
+
+OpenLineage spec has already contained for some time [OutputStatisticsDatasetFacet](https://openlineage.io/docs/spec/facets/dataset-facets/output-dataset-facets/output_statistics/).
+Recently, we added [InputStatisticsDatasetFacet](https://openlineage.io/docs/spec/facets/dataset-facets/input-dataset-facets/input_statistics/).
+Both facets contain basic statistics like number of rows processed, size in bytes, and file count.
+
+We can demonstrate them in action by running Jupyter quickstart demo which emit OpenLineage events to
+docker logs. We start Spark session with:
+```python
+from pyspark.sql import SparkSession
+
+spark = (SparkSession.builder.master('local')
+         .appName('sample_spark')
+         .config('spark.jars.packages', 'org.apache.iceberg:iceberg-spark-runtime-3.5_2.12:1.7.1')
+         .config('spark.jars', '/home/jovyan/openlineage/libs/shadow/openlineage-spark-agent_2.12-1.26.0-SNAPSHOT-shadow.jar')
```

Comment:
I know the `jar` hasn't been released yet, but maybe add a reference to the (future?) maven artifact vs a local path? 

Or:

```
/path/to/openlineage/libs/shadow/openlineage-spark-agent_2.12-1.26.0-SNAPSHOT-shadow.jar
```

Reply:
switching to a released version